language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -439,7 +443,18 @@ public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormatting
                 if (syntaxTree.isEmpty()) {
                     return Collections.emptyList();
                 }
-                String formattedSource = Formatter.format(syntaxTree.get()).toSourceCode();
+                String formattedSource;
+                if (context.currentModule().isPresent() &&
+                        !(context.currentModule().get().project() instanceof SingleFileProject)) {
```

Comment:
Shall we use project `kind()` instead of `instanceof` check?
We need to avoid `instanceof` checks everywhere possible, as they are slow.